### DISCUSSION TITLE
Getting element in page source but appium is not able to perform the click.

### DISCUSSION THREAD

**Discussion by attrideepak** [appium/appium#17001] [Q&A] [answered]
The element is present in pageSource but when appium tries to click it throws noSuchElementException.
**Exception**
```
Original error: java.util.NoSuchElementException: The element identified by 'c4b50604-101f-4081-ba7d-c095723c62fc' does not exist in the cache or has expired. Try to find it again
Build info: version: '4.1.2', revision: '9a5a329c5a'
```

**Comment by attrideepak** [comment, marked as answer]
Before clicking on the element I am checking `element.isEnabled()` and `element.isDisplayed()` which is not working with jetpack compose elements and hence the exception, but not sure why these methods do not work with jetpack compose elements. I have removed this check and it works fine now.
Any possible explanation ?

**Reply by KazuCocoa** [reply]
The endpoint calls compose's apis, so potentially the compose element did not respond its expected values.

- https://github.com/appium/appium-espresso-driver/blob/08b877b47d2c826f3059ce63f2303bed329fc458/espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/ComposeNodeElement.kt#L65
    - https://developer.android.com/reference/kotlin/androidx/compose/ui/semantics/SemanticsConfiguration
- androidx.compose.ui.test.assertIsDisplayed
    - https://github.com/appium/appium-espresso-driver/blob/9204a7ea6f12015ad9611cdf1b596289a8c4da52/espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetDisplayed.kt#L32

**Reply by attrideepak** [reply]
Thanks for the reply. I tried the same scenario with `element.getAttribute("enabled")` and it works fine, doesn't both these methods i.e. `isEnabled()` and `getAttribute("enabled")` calls same API?

**Reply by KazuCocoa** [reply]
https://github.com/appium/appium-espresso-driver/blob/08b877b47d2c826f3059ce63f2303bed329fc458/espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/ComposeNodeElement.kt#L110
I read the code. Probably no.

https://github.com/appium/appium-espresso-driver/blob/006bcfcf5a4d07713cb3e6d7a032ce80704cf28d/espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetEnabled.kt#L25 referred only espresso element

**Reply by KazuCocoa** [reply]
Could you try out https://github.com/appium/appium-espresso-driver/pull/784 ?

**Reply by attrideepak** [reply]
I can try, but tell me how can I have these changes in my machine?

**Reply by KazuCocoa** [reply]
the easiest way is to replace `espresso-server` directory in `.appium/node_modules/appium-espresso-driver/espresso-server` with the PR branch's `espresso-server`.

**Reply by attrideepak** [reply]
verfied the fix.. works now. Thanks for the support.

**Reply by KazuCocoa** [reply]
2.4.0 Espresso driver has the change.
